packages/frontend/model/age-warning.ts

```diff
@@ -0,0 +1,43 @@
+export const getAgeWarning = (
+    tags: TagType[],
+    webPublicationDate: string,
+): string | undefined => {
+    const isNews = tags.some(t => t.id === 'tone/news');
+
+    if (!isNews) {
+        return;
+    }
+
+    const warnLimitDays = 30;
+    const currentDate = new Date();
+    const dateThreshold = new Date();
+
+    dateThreshold.setDate(currentDate.getDate() - warnLimitDays);
+
+    const publicationDate = new Date(webPublicationDate);
+
+    // if the publication date is before the date threshold generate message
+    if (publicationDate < dateThreshold) {
+        // Unary + coerces dates to numbers for TypeScript
+        const diffMilliseconds = +currentDate - +publicationDate;
```

Comment:
TIL

Reply:
Ha I hadn't even noticed that (this is copy + pasted stuff).